src/baldr/tilehierarchy.cc

```diff
@@ -1,44 +1,100 @@
 #include "baldr/tilehierarchy.h"
 #include "baldr/graphtileheader.h"
+#include "boost/foreach.hpp"
+#include "config.h"
 #include "midgard/vector2.h"
+#include <cassert>
 
 using namespace valhalla::midgard;
 
 namespace valhalla {
 namespace baldr {
 
-const std::vector<TileLevel>& TileHierarchy::levels() {
-  // Static tile levels
-  static const std::vector<TileLevel> levels_ = {
+static std::vector<TileLevel> levels_;
+static std::vector<TileLevel> transit_levels_;
+
+void getLevelsfromConfig() {
+  std::vector<float> tilesizesvector;
+  bool bSizesInConfigOk = false;
+  const AABB2<midgard::PointLL> wholemapbb{
+      {-180., -90.},
+      {180., 90.},
+  };
+
+  try {
+    // Many tests don't specify any configuration, so catch the "Config singleton was not initialized
+    // before usage" exception here and fallback to default OSM tile grid.
+    const boost::optional<const boost::property_tree::ptree&> tilesizesptree =
+        config().get_child_optional("baldr.tilesizes");
+
+    if (tilesizesptree) {
+      BOOST_FOREACH (auto& v, *tilesizesptree) { tilesizesvector.push_back(v.second.get<float>("")); }
```

Comment:
modern c++ has foreach for a long time, i know ptree is a pain but i think it should work with:

```suggestion
    for(auto& v : *tilesizesptree) { 
      tilesizesvector.push_back(v.second.get<float>(""));
    }
```